src/service/riskScoring.js

```diff
@@ -0,0 +1,35 @@
+import Fingerprint from 'fingerprintjs2';
+import request from '@/class/singleton/request';
+
+/**
+ * @param {object[]} entries
+ * @returns {void}
+ */
+const sendFingerprint = () => {
+  const finger = new Fingerprint({
+    excludeCanvas: true,
+  });
+
+  finger.get(async (hash, entries) => {
```

Comment:
I think, better to make it as promise because we are working with remote server and should handle all errors and issues

Reply:
if you about `get`, what's the sense? server part depends of `request` class, so we just can wrap `request` to try \ catch and rethrow exception upwards

Reply:
Under the hood we have a promise and it is not obvious on usage level - this can produce some unessesary side effects

Reply:
there are haven't a way to get promise

Reply:
you can use `const components = await Fingerprint.getPromise(options)` without callback

Reply:
Nope, `getPromise` was added only in new versions (2.x.x), in which there aren't way to handle like a module. So we use older version, which can be included as a module, but without promise.

I guess this problem is not critical, because there aren't fundamental differences between resulting of promises and callbacks. `await` just syntax sugar, under the hood this is typical promise, and typical promise is pretty similar (and works the same) like common function that calls their callbacks when it needed. No magic and miracles here.

Reply:
so, better to separate local method
```
async getFingerJson() {
 return new Promise((resolve) =>{
   finger.get((hash, entries) => {
    resolve(entries.reduce(
      (accumulator, entry) => ({
        [entry.key]: entry.value,
        ...accumulator,
      }),
      {},
    ));
   });
}));
}
```
and then just call `const json = await this.getFingerJson()`;

because callbacks is not our way

Reply:
and why we can't use 2.x version?

Reply:
> and why we can't use 2.x version?

there aren't way to work with library like a module. So we use older version, which can be included as a module, but without promise.